google/cloud/bigtable/row_data.py

```diff
@@ -625,16 +626,11 @@ def __init__(self, message, last_scanned_key, rows_read_so_far):
     def build_updated_request(self):
         """Updates the given message request as per last scanned key"""
 
-        # TODO: Generalize this to ensure fields don't get rewritten when retrying the request
-
-        r_kwargs = {
-            "table_name": self.message.table_name,
-            "filter": self.message.filter,
-            "app_profile_id": self.message.app_profile_id,
-        }
+        updated_message = data_messages_v2_pb2.ReadRowsRequest()
+        proto.Message.copy_from(updated_message, self.message)
 
         if self.message.rows_limit != 0:
-            r_kwargs["rows_limit"] = max(
+            updated_message.rows_limit = max(
                 1, self.message.rows_limit - self.rows_read_so_far
```

Comment:
why max? If the `rows_limit - self.rows_read_so_far` is negative then something is terribly wrong and we should throw an error. If its 0 then then stream should be completed before we even get here. Please add a TODO: to address this.

Reply:
I've added the TODO, and the tests have been fixed.